### PR TITLE
NoPartialFunctions: detect instance members Option.Value, Map.Item, List.Head & List.Item

### PR DESCRIPTION
**Pull request by su8898**

### CODE REVIEW

File: src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs [hunk 1]
```diff
@@ -67,6 +70,15 @@ let private partialFunctionIdentifiers =
         ("List.pick", Function "List.tryPick")
     ] |> Map.ofList
 
+let private partialInstanceMemberIdentifiers =
+    [
+        ("Option.Value", PatternMatch)
+        ("Map.Item", Function "Map.tryFind")
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs:76]: let's add also List.Head -> List.tryHead and List.Item -> List.tryItem

**Comment by su8898** [comment]
Hi @knocte, I've updated the PR with `List.Head` and `List.Item`

**Comment by knocte** [comment]
> I've updated the PR with List.Head and List.Item

Nit: you need to update PR title and commit message then.

**Comment by su8898** [comment]
> 

I've updated the commit message and PR title to reflect the addition of List.Head & List.Item

File: src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs [hunk 2]
```diff
@@ -96,13 +110,92 @@ let private checkIfPartialIdentifier (config:Config) (identifier:string) (range:
...
+            match replacement with
+            | (fullyQualifiedInstanceMember, replacementStrategy) ->
+                if not (fullyQualifiedInstanceMember.Contains ".") then
+                    failwith "Please use fully qualified name for the instance member"
+                let nameSegments = fullyQualifiedInstanceMember.Split '.'
+                let instanceMemberNameOnly = Array.last nameSegments
+                let isSourcePropSameAsReplacementProp = List.tryFind (fun sourceInstanceMemberName -> sourceInstanceMemberName = instanceMemberNameOnly) names
+                match isSourcePropSameAsReplacementProp with
+                | Some _ ->
+                    let typeName = fullyQualifiedInstanceMember.Substring(0, fullyQualifiedInstanceMember.Length-instanceMemberNameOnly.Length-1)
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs:128]: put spaces around the `-` operator please

Comment (su8898) [reply src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs]: @knocte thanks. I've updated the PR with this fix.

**Pull request merged**